pages/api/v1/social.js

```diff
@@ -0,0 +1,48 @@
+import * as Data from "~/node_common/data";
+import * as Serializers from "~/node_common/serializers";
+
+export default async (req, res) => {
+  const subscriptions = await Data.getSubscriptionsByUserId({ userId: req.body.data.userId });
+  const subscribers = await Data.getSubscribersByUserId({ userId: req.body.data.userId });
+
+  let serializedUsersMap = { [req.body.data.userId]: req.body.data };
+  let serializedSlatesMap = {};
+
+  const r1 = await Serializers.doSubscriptions({
+    users: [],
+    slates: [],
+    subscriptions,
+    serializedUsersMap,
+    serializedSlatesMap,
+  });
+
+  const r2 = await Serializers.doSubscribers({
+    users: [],
+    slates: [],
+    subscribers,
+    serializedUsersMap: r1.serializedUsersMap,
+    serializedSlatesMap: r1.serializedSlatesMap,
+  });
+
+  if (!subscriptions) {
```

Comment:
Would be better to cancel the request sooner if the first one failed., the decorator looks like its copy and pasted and should be specific to this call

Reply:
@tarafanlin so when you're working with the backend code, you want to send different decorators for each error so that you know which error is happening when you get an error back. Take a look at the other files to get an idea of what I mean. 

The other thing is whenever you add a new decorator, you also should go to common/messages.js and add a corresponding message. This is how know which error message to show the user: when the frontend receives an error back from an api call, it passes it through this Events.hasError(response) function which triggers an alert if there is an error. That alert looks up the decorator in messages.js, and shows the corresponding alert message (or a generic alert message if not found)

Reply:
oh I finally see what you guys mean. 
the english word "subscriptions" and "subscribers" look too similar to me, I didn't realize the two are using the same decorator.